<launch>
	<arg name="topic_of_local_pcl" default="/prometheus/sensors/2Dlidar_scan"/>
	<!-- 启动局部规划算法 -->
	<node pkg="prometheus_local_planner" name="local_planner_main" type="local_planner_main" output="screen" launch-prefix="gnome-terminal --">	
		<remap from="/prometheus/planning/local_pcl" to="$(arg topic_of_local_pcl)" />
		<!-- 选择算法 0代表APF,1代表VFH -->
		<param name="local_planner/algorithm_mode" value="1" type="int"/>
		<!-- 激光雷达模型,0代表3d雷达,1代表2d雷达 -->
		<param name="local_planner/lidar_model" value="1" type="int"/>
		<!-- 飞行高度 -->
		<param name="local_planner/fly_height_2D" value="1.0" type="double"/>
		<param name="local_planner/sim_mode" value="true" type="bool"/>
		<!-- 最大速度，设的越小越安全 -->
		<param name="local_planner/max_planning_vel" value="0.5" type="double"/>
		<!-- 膨胀参数，一般设置为无人机的半径或更大 -->
		<param name="vfh/inflate_distance" value="1.0" type="double"/>
		<!-- 感知距离，只考虑感知距离内的障碍物 -->
		<param name="vfh/sensor_max_range" value="6.0" type="double"/>  
		<!-- weight -->
		<param name="vfh/obstacle_weight" value="10.0" type="double"/>
		<param name="vfh/goalWeight" value="0.1" type="double"/>
		<param name="vfh/safe_distance" value="0.01" type="double"/>
		<!-- 直方图个数 -->
		<param name="vfh/h_res" value="360" type="int"/>
	</node>	
</launch>